### ISSUE TITLE
Upgrading yarn editor to the newest version

### ISSUE THREAD

**Issue by blurymind** [4ian/GDevelop#6460]
## Description
I am wondering if its worth making a pr to upgrade yarn editor to a newer version.
There are some missing bits and fixed bugs/ improvement of life features in the newest version. Is this something that would be welcome as a Pr?

Improvements over the old one:
- much better performance with big big files
- support for ink story file format import and export + syntax (possibly useful for the future)
- better handling of touch input and small size screens (smartphones/tablets)
- bug fixes

## Solution suggested

Before doing a PR I think I will do some work on yarn editor to trim it down - to reduce its size. It has grown in size due to bundling of an inkjs compiler and some word library files - to enable previewing ink stories directly inside yarn + suggestion of related words to selected word. 

I am planning to trim it down and speed up its start time, so the ink playtesting feature might be reworked or removed, but the optional ink import/export/syntax support will stay

**Comment by blurymind** [comment]
Might give this one higher priority than upgrading bondagejs, seeing as to how its getting more hearts.

I did some work on yarn editor to trim down its bundle size while keeping as many of the new features as possible over the weekend.
I can also say that it handles giant files (over 500+ nodes with lots  of text) much much more efficiently than what is currently used in Gdevelop.

There was some badly optimised stuff. Like typing to search for a node was very slow, panning was slow, loading the file was slow

**Comment by MTSyntho** [comment]
I think the implementation of the new yarn editor + being able to handle tons of nodes will be incredibly useful for my upcoming via move, would contribute if i really understood js, ts & such much more but sadly can't, still tho keep up the great work 🗣️📢🔥

**Comment by blurymind** [comment]
yup I understand the demand. The good news is that I managed to trim down the newest version of yarn to almost the same size as it is now in gdevelop's old version, while retaining all the new the features.

I remember when I initially did the bundle pr - the deploy size being small was a very important point.
There are some api changes that came with turning it into a pwa, which may need proper re-testing in its wrapper in gdevelop

I want to sort out my vn actor pr before doing the yarn editor upgrade though
https://github.com/4ian/GDevelop/pull/6469
